src/stripe/stripe-service.ts

```diff
@@ -322,4 +344,90 @@ export class StripeService {
       confirmationTokenId: confirmationToken.id,
     };
   }
+
+  static nextDateforPeriod(period: Period, startDate: Date) {
+    if (period.unit === PeriodUnit.Year) {
+      startDate.setFullYear(startDate.getFullYear() + period.number);
+      return startDate;
+    }
+
+    if (period.unit === PeriodUnit.Month) {
+      startDate.setMonth(startDate.getMonth() + period.number);
+      return startDate;
+    }
+
+    if (period.unit === PeriodUnit.Week) {
+      startDate.setDate(startDate.getDate() + period.number * 7);
+      return startDate;
+    }
+
+    if (period.unit === PeriodUnit.Day) {
+      startDate.setDate(startDate.getDate() + period.number);
+      return startDate;
+    }
+
+    return startDate;
+  }
+
+  static applePayPeriod(period: Period): {
+    recurringPaymentIntervalUnit?: "year" | "month" | "day" | "hour" | "minute";
+    recurringPaymentIntervalCount?: number;
+  } {
+    if (period.unit === PeriodUnit.Week) {
+      return {
+        recurringPaymentIntervalUnit: "day",
+        recurringPaymentIntervalCount: period.number * 7,
+      };
+    }
+    return {
+      recurringPaymentIntervalUnit: period.unit,
+      recurringPaymentIntervalCount: period.number,
+    };
+  }
+
+  static buildStripeExpressCheckoutOptionsForSubscription(
+    productDetails: Product,
+    subscriptionOption: SubscriptionOption,
+    translator: Translator,
+  ): StripeExpressCheckoutOptions {
+    const priceMicros = subscriptionOption.base.price?.amountMicros;
+
+    if (!priceMicros) {
+      return {};
+    }
+
+    const hasTrial = subscriptionOption.trial;
+    const trialPeriod = subscriptionOption.trial?.period;
+    const basePeriod = subscriptionOption.base.period;
+
+    const recurringPaymentStartDate =
+      hasTrial && trialPeriod
+        ? StripeService.nextDateforPeriod(trialPeriod, new Date())
+        : new Date();
+
+    const recurringPeriod = basePeriod
+      ? StripeService.applePayPeriod(basePeriod)
+      : {};
+
+    return {
+      applePay: {
+        recurringPaymentRequest: {
+          paymentDescription: productDetails.title,
+          managementURL: "https://example.com/manage",
```

Comment:
This looks off, we should build a proper management URL here using the `op_session_id`

Reply:
Yep, working on sending it from Khepri